tests/_internal/clients/test_alertrules.py

```diff
@@ -0,0 +1,142 @@
+        alert_rule_client = AlertRulesClient(mock_session, user_context)
+        alert_rule_client.remove_all_users(u"rule-id")
+
+        assert mock_session.post.call_count == 1
+        posted_data = json.loads(mock_session.post.call_args[1]["data"])
+        assert mock_session.post.call_args[0][0] == "/svc/api/v1/Rules/remove-all-users"
+        assert (
+            posted_data["tenantId"] == user_context.get_current_tenant_id()
+            and posted_data["ruleId"] == u"rule-id"
+        )
+
+    def test_get_by_id_posts_expected_data_for_exfiltration_type(self, mock_session, user_context):
+        alert_rule_client = AlertRulesClient(mock_session, user_context)
+        alert_rule_client.get("exfiltration", u"rule-id")
+
+        assert mock_session.post.call_count == 1
+        posted_data = json.loads(mock_session.post.call_args[1]["data"])
+        url = mock_session.post.call_args[0][0]
+        assert url == "/svc/api/v1/Rules/query-endpoint-exfiltration-rule"
+        assert posted_data["tenantId"] == user_context.get_current_tenant_id() and posted_data[
+            "ruleIds"
+        ] == [u"rule-id"]
+
+    def test_get_by_id_posts_to_correct_endpoint_for_cloudshare_type(
+        self, mock_session, user_context
+    ):
+        alert_rule_client = AlertRulesClient(mock_session, user_context)
+        alert_rule_client.get("cloudshare", u"rule-id")
+        url = mock_session.post.call_args[0][0]
+        assert url == "/svc/api/v1/Rules/query-cloud-share-permissions-rule"
+
+    def test_get_by_id_posts_to_correct_endpoint_for_type_mismatch_rule_type(
+        self, mock_session, user_context
+    ):
+        alert_rule_client = AlertRulesClient(mock_session, user_context)
+        alert_rule_client.get("typemismatch", u"rule-id")
+        url = mock_session.post.call_args[0][0]
+        assert url == "/svc/api/v1/Rules/query-file-type-mismatch-rule"
+
+
+class TestAlertRulesManagerClient(object):
+    def test_get_all_posts_expected_data(self, mock_session, user_context):
+        alert_rule_client = AlertRulesManagerClient(mock_session, user_context)
+
+        for _ in alert_rule_client.get_all(sort_key="key", sort_direction="ASC"):
+            break
+
+        assert mock_session.post.call_count == 1
+        posted_data = json.loads(mock_session.post.call_args[1]["data"])
+        assert mock_session.post.call_args[0][0] == "/svc/api/v1/rules/query-rule-metadata"
+        assert (
+            posted_data["tenantId"] == user_context.get_current_tenant_id()
+            and posted_data["groups"] == []
+            and posted_data["groupClause"] == "AND"
+            and posted_data["pgNum"] == 0
+            and posted_data["pgSize"] == 1000
+            and posted_data["srtKey"] == "key"
+            and posted_data["srtDirection"] == "ASC"
+        )
+
+    @pytest.mark.skip("TODO add assert statements")
+    def test_get_by_name_posts_expected_data(self, mocker, py42_response):
```

Comment:
Need help to figure out how to test this. `get_all` returns a generator containing `py42Response`, so instead of generator I have set its return value to a list. However this doesn't work for me. 

I want to test that the code indeed fetches the record with given name.